### PR TITLE
[fix] Refactored Dockerfiles

### PR DESCRIPTION
**Pull request by adriang-90**
**Description**
Refactored **Dockerfiles** using best practices from official docker docs and many blog posts. 
* Changed **latest** tag in image declaration -> https://platformers.dev/log/2022-03-02-latest-literally-kills-puppies/
* Added **WORKDIR** before **COPY** in final stages for more readability
* Changed **ADD** FOR **COPY** since you do not copy any *.tar files -> [nr 7](https://res.cloudinary.com/snyk/image/upload/v1551798390/Docker_Image_Security_Best_Practices_.pdf)
* added **yarn cache clean** -> https://github.com/hadolint/hadolint/issues/395

**Additional context**
* With **hadolint** I made some changes that allowed me to obtain less disk space usage:
* Before
```
REPOSITORY                                 TAG       IMAGE ID       CREATED          SIZE
ethereumoptimism/deployer                  latest    07e939226ecc   6 minutes ago    1.85GB
ethereumoptimism/message-relayer           latest    7d19b23f930d   6 minutes ago    1.85GB
ethereumoptimism/integration-tests         latest    5963071e94cc   6 minutes ago    1.85GB
ethereumoptimism/data-transport-layer      latest    cb2ab0c2728e   6 minutes ago    1.85GB
ethereumoptimism/l2geth                    latest    d98fec8af817   8 minutes ago    43.8MB
ethereumoptimism/batch-submitter-service   latest    0b928130c226   9 minutes ago    29MB
ethereumoptimism/gas-oracle                latest    e9960474419b   10 minutes ago   22MB
ethereumoptimism/hardhat                   latest    790ade562afe   11 minutes ago   298MB
```
* after
```
REPOSITORY                                 TAG       IMAGE ID       CREATED              SIZE
ethereumoptimism/integration-tests         latest    9f9e87922547   About a minute ago   1.12GB
ethereumoptimism/deployer                  latest    d5f7def7f0ce   About a minute ago   1.12GB
ethereumoptimism/message-relayer           latest    6d06b3551059   About a minute ago   1.12GB
ethereumoptimism/data-transport-layer      latest    516f0e77af55   About a minute ago   1.12GB
ethereumoptimism/gas-oracle                latest    4e030f098cd2   About an hour ago    22MB
ethereumoptimism/batch-submitter-service   latest    6f5ae41cea21   About an hour ago    29MB
ethereumoptimism/l2geth                    latest    d42d4f23fe32   About an hour ago    43.8MB
ethereumoptimism/hardhat                   latest    60c00a3273f8   9 seconds ago       193MB
```

### CODE REVIEW

**Comment by tynes** [comment]
Thank you for the PR! This is very helpful. Good looks on pinning the version of alpine. Just approved CI to run, if everything builds fine then I'm open to approving it. Ideally there is a changeset for each of:

- integration-tests
- contracts
- message-relayer
- data-transport-layer
- gas-oracle
- batch-submitter-service
- l2geth
- hardhat

Each should be a patch release, this will trigger builds of and pushes to dockerhub when we do decide to do a release of the software. To add changesets, run `yarn changeset` from the root of the repo.

**Comment by adriang-90** [comment]
This command does not seem to work for me, being in root directory:
```
yarn run v1.22.17
error Command "changeset" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Am I doing something wrong?

**Comment by tynes** [comment]
Perhaps you need to `yarn install`? The `changeset` executable is `node_modules/.bin/changeset`. If you would prefer me to add them, I can do so

**Comment by adriang-90** [comment]
Yes you got the point. I added changeset.

File: .changeset/mighty-news-jam.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+'@eth-optimism/batch-submitter-service': patch
```
Comment (smartcontracts) [inline .changeset/mighty-news-jam.md:2]: Seems like there's a duplicated changeset here

**Review by smartcontracts — COMMENTED** [review]
Just need to combine into one commit via rebase and remove the extra changeset, then should be good to go!

**Review by mslipper — APPROVED** [review]

**Pull request merged**